opentelemetry/src/sdk/trace/span_processor.rs

```diff
@@ -104,13 +103,37 @@ pub trait SpanProcessor: Send + Sync + std::fmt::Debug {
 /// ```
 #[derive(Debug)]
 pub struct SimpleSpanProcessor {
-    exporter: Mutex<Box<dyn SpanExporter>>,
+    sender: crossbeam_channel::Sender<Option<SpanData>>,
+    shutdown: crossbeam_channel::Receiver<()>,
 }
 
 impl SimpleSpanProcessor {
-    pub(crate) fn new(exporter: Box<dyn SpanExporter>) -> Self {
+    pub(crate) fn new(mut exporter: Box<dyn SpanExporter>) -> Self {
+        let (span_tx, span_rx) = crossbeam_channel::unbounded();
+        let (shutdown_tx, shutdown_rx) = crossbeam_channel::bounded(0);
+
+        let _ = thread::Builder::new()
+            .name("opentelemetry-exporter".to_string())
+            .spawn(move || {
+                while let Ok(Some(span)) = span_rx.recv() {
+                    if let Err(err) = executor::block_on(exporter.export(vec![span])) {
```

Comment:
Should we consider adding a timeout around here to prevent the exporter block for too long?

Reply:
I was considering that, but the spec doesn't have any config for it and if you are creating faster than you are exporting to the point where it would be a big issue you should probably just use the batch processor as your app may OOM or do other bad things as well with the simple processor.

Reply:
To that point we should probably warn users that using simple span processor may cause OOM.

Reply:
Yeah I think a docs PR followup on the difference between both and when to choose one vs the other would be a good idea 👍